integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,424 @@
+         * @param metadata metric metadata describing the metric
+         * @return the formatted metric name
+         */
+        default String format(MetricID metricId, String suffix, Metadata metadata) {
+
+            MetricType metricType = metadata.getTypeRaw();
+
+            StringBuilder result = new StringBuilder(metricId.getName());
+            if (suffix != null) {
+                result.append("_").append(suffix);
+            }
+            result.append("_").append(metricType);
+
+            String units = formattedBaseUnits(metadata.getUnit().orElse(null));
+            if (units != null && !units.isBlank()) {
+                result.append("_").append(units);
+            }
+            return result.toString();
+        }
+    }
+
+    static String formattedBaseUnits(String metricUnits) {
+        String baseUnits = baseMetricUnits(metricUnits);
+        return baseUnits == null ? "" : baseUnits;
+    }
+
+    static String baseMetricUnits(String metricUnits) {
+        if (!MetricUnits.NONE.equals(metricUnits) && !metricUnits.isEmpty()) {
+            for (UnitConverter converter : UNIT_CONVERTERS) {
+                if (converter.handles(metricUnits)) {
+                    return converter.baseUnits();
+                }
+            }
+        }
+        return null;
+    }
+
+    private void shutdown(WebServer webServer) {
+        if (webServerCounter.decrementAndGet() < 1) {
+            LOGGER.fine("Shutting down OCI Metrics agent");
+            scheduledExecutorService.shutdownNow();
+        }
+    }
+
+    private void startExecutor() {
+        scheduledExecutorService = Executors.newScheduledThreadPool(1);
+        scheduledExecutorService.scheduleAtFixedRate(this::pushMetrics, initialDelay, delay, schedulingTimeUnit);
+    }
+
+    @Override
+    public void update(Routing.Rules rules) {
+        rules.onNewWebServer(this::prepareShutdown);
+
+        if (!enabled) {
+            LOGGER.info("Metric push to OCI is disabled!");
+            return;
+        }
+
+        if (scopes.length == 0) {
+            LOGGER.info("No selected metric scopes to push to OCI");
+            return;
+        }
+
+        LOGGER.fine("Starting OCI Metrics agent");
+
+        RegistryFactory rf = RegistryFactory.getInstance();
+        Stream.of(scopes)
+                .forEach(type -> metricRegistries.put(rf.getRegistry(type), type));
+
+        ociMetricsData = new OciMetricsData(
+                metricRegistries, nameFormatter, compartmentId, namespace, resourceGroup, descriptionEnabled);
+        startExecutor();
+    }
+
+    private void prepareShutdown(WebServer webServer) {
+        webServerCounter.incrementAndGet();
+        webServer.whenShutdown().thenAccept(this::shutdown);
+    }
+
+    /**
+     * Builder for OciMetricsSupport.
+     */
+    public static class Builder implements io.helidon.common.Builder<OciMetricsSupport> {
```

Comment:
This builder class needs the `@Configured` annotation.

Reply:
Updated in the most recent commit.